app/src/org/commcare/android/view/EntityView.java

```diff
@@ -530,5 +491,8 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
                 views[i].setLayoutParams(params);
             }
         }
-    }*/
+        
+        // Use children's new widths
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);

```

Comment:
The first `super.onMeasure` is so that `getMeasuredWidth` returns the correct width of the row. The function then sets all of the children's desired widths, and the second `super.onMeasure` is where the children's new dimensions actually get set.

^^ This was a "take stabs in the dark until something works" kind of bug, followed by looking at https://developer.android.com/guide/topics/ui/how-android-draws.html to rationalize *why* it works.